$MirOS$

	• support for crtbeginT.o
	• support for -fstack-protector-all (alias to -fstack-protector)
	• fix bogus if location for if STARTFILES_S is not defined
	• crti.o MUST BE the first thing on the ld(1) command line, thus
	  move startfiles before crt0file

--- cc/cc/cc.c.orig	Thu Jan  1 00:00:00 1970
+++ cc/cc/cc.c	Thu Jul 17 21:13:06 2008
@@ -200,6 +200,9 @@ char *endfiles[] = ENDFILES;
 char *startfiles_S[] = STARTFILES_S;
 char *endfiles_S[] = ENDFILES_S;
 #endif
+#ifdef STARTFILES_T
+char *startfiles_T[] = STARTFILES_T;
+#endif
 char *cppmdadd[] = CPPMDADD;
 #ifdef LIBCLIBS
 char *libclibs[] = LIBCLIBS;
@@ -327,6 +330,10 @@ main(int argc, char *argv[])
 					flist[nf++] = argv[i];
 					sspflag++;
 				} else if (strcmp(argv[i],
+				    "-fstack-protector-all") == 0) {
+					flist[nf++] = argv[i];
+					sspflag++;
+				} else if (strcmp(argv[i],
 				    "-fno-stack-protector") == 0) {
 					flist[nf++] = argv[i];
 					sspflag = 0;
@@ -731,13 +738,26 @@ nocom:
 			av[j++] = "-o";
 			av[j++] = outfile;
 		}
-		if (shared) {
 #ifdef STARTFILES_S
+		if (shared) {
 			for (i = 0; startfiles_S[i]; i++)
 				av[j++] = startfiles_S[i];
+		} else
 #endif
-		} else {
+		{
 			if (!nostartfiles) {
+#ifdef STARTFILES
+#ifdef STARTFILES_T
+				if (Bstatic) {
+					for (i = 0; startfiles_T[i]; i++)
+						av[j++] = startfiles_T[i];
+				} else
+#endif
+				{
+					for (i = 0; startfiles[i]; i++)
+						av[j++] = startfiles[i];
+				}
+#endif
 #ifdef CRT0FILE_PROFILE
 				if (pgflag)
 				{
@@ -750,10 +770,6 @@ nocom:
 					av[j++] = crt0file;
 #endif
 				}
-#ifdef STARTFILES
-				for (i = 0; startfiles[i]; i++)
-					av[j++] = startfiles[i];
-#endif
 			}
 		}
 		i = 0;
